src/solver/algorithm/strong/cyclic.rs

```diff
@@ -6,3 +6,366 @@
+                    losing_frontier.push_back(parent);
+                }
+                // Update child count
+                child_counts.insert(parent, child_count - 1);
+            }
+        }
+        // Child should never be a tying position
+        else if matches!(child_utility, SimpleUtility::TIE) {
+            for parent in parents {
+                let child_count = *child_counts.get(&parent).expect(
+                    "Failed to enqueue parent state in initial enqueuing stage",
+                );
+                // Parent has already been solved
+                if child_count == 0 {
+                    continue;
+                }
+                // Add database entry
+                let mut buf = RecordBuffer::new(game.players())
+                    .context("Failed to create record for end state.")?;
+                buf.set_utility([SimpleUtility::TIE, SimpleUtility::TIE])?;
+                buf.set_remoteness(child_remoteness + 1)?;
+                db.put(parent, &buf);
+
+                // Add parent to win frontier
+                tying_frontier.push_back(parent);
+                // Update child count
+                child_counts.insert(parent, 0);
+            }
+        } else {
+            panic!["Position with invalid utility found in frontiers"];
+        }
+    }
+
+    // Assign drawing utility
+    for (parent, child_count) in child_counts {
+        if child_count > 0 {
+            let mut buf = RecordBuffer::new(game.players())
+                .context("Failed to create record for end state.")?;
+            buf.set_utility([SimpleUtility::DRAW, SimpleUtility::DRAW])?;
+            db.put(parent, &buf);
+        }
+    }
+
+    Ok(())
+}
+
+/// Set up the initial frontiers and primitive position database entries
+fn enqueue_children<G, D>(
+    winning_frontier: &mut VecDeque<State>,
+    tying_frontier: &mut VecDeque<State>,
+    losing_frontier: &mut VecDeque<State>,
+    curr_state: State,
+    game: &G,
+    child_counts: &mut HashMap<State, usize>,
+    db: &mut D,
+) -> Result<()>
+where
+    G: DTransition<State> + Bounded<State> + SimpleSum<2> + Extensive<2> + Game,
+    D: KVStore,
+{
+    if game.end(curr_state) {
+        let mut buf = RecordBuffer::new(game.players())
+            .context("Failed to create placeholder record.")?;
+        buf.set_utility(game.utility(curr_state))
+            .context("Failed to copy utility values to record.")?;
+        buf.set_remoteness(0)
+            .context("Failed to set remoteness for end state.")?;
+        db.put(curr_state, &buf);
+
+        match game
+            .utility(curr_state)
+            .get(game.turn(curr_state))
+        {
+            Some(SimpleUtility::WIN) => winning_frontier.push_back(curr_state),
+            Some(SimpleUtility::TIE) => tying_frontier.push_back(curr_state),
+            Some(SimpleUtility::LOSE) => losing_frontier.push_back(curr_state),
+            _ => {
```

Comment:
Avoid using `_` in `match` statements, opt instead to be exhaustive (for the same reason as I previously explained).